subsys/bluetooth/services/fast_pair/fmdn/state.c

```diff
@@ -1295,14 +1297,12 @@ int fp_fmdn_state_init(void)
 		return err;
 	}
 
-	/* Create advertising set if beacon is provisioned. */
-	if (is_provisioned) {
-		if (!fmdn_adv_set) {
-			err = fmdn_adv_set_setup();
-			if (err) {
-				LOG_ERR("FMDN State: fmdn_adv_set_setup failed: %d", err);
-				return err;
-			}
+	/* Reserve the FMDN advertising set only in the Fast Pair enabled state. */
+	if (!fmdn_adv_set) {
```

Comment:
`bt_fast_pair_fmdn_tx_power_verify` asserts that there is no `fmdn_adv_set` which is not consistent with approach taken here. Since we always uninitialize and delete `fp_adv_set` on init failure, can we convert the if check to an assertion here?

Or maybe we could instead reuse the `fp_adv_set` for TX power validation to avoid problems on adv set delete failure during uninit (we implicitly validate TX power while creating FMDN adv set anyway so maybe the duplicated validation logic could simply be removed).